### PR TITLE
add GET /api/articles/:article_id with tests

### PR DESCRIPTION
**Pull request by Slyose**

### CODE REVIEW

**Comment by Slyose** [comment]
Realized i left a few console.logs, updating it

File: __tests__/app.test.js [hunk 1]
```diff
@@ -46,15 +58,39 @@ describe("GET /api/topics", () => {
...
+        expect(article.votes).toEqual(expect.any(Number));
+      });
+  });
+  test("Should return a 404 and 'article_id does not exist' when given an invalid article_ID", () => {
+    return request(app)
+      .get("/api/articles/9001")
+      .expect(404)
+      .then(({ _body }) => {
+        expect(_body.msg).toBe("article_id does not exist");
+      });
```
Comment (cjplatten) [inline __tests__/app.test.js:97]: Can you think of any other potential errors for this endpoint?

**Comment by cjplatten** [comment]
Just one comment to address, otherwise looking good!

File: __tests__/app.test.js (hunk 1 above)
Comment (Slyose) [reply __tests__/app.test.js:97]: Yep, added error handling for a bad request (400)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const request = require("supertest");
 const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
 const data = require("../db/data/test-data");
+const { response } = require("../app");
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: this line is unnecessary. What is response?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -46,15 +58,47 @@ describe("GET /api/topics", () => {
...
+        expect(article.body).toEqual(expect.any(String));
+        expect(article.created_at).toEqual(expect.any(String));
+        expect(article.votes).toEqual(expect.any(Number));
+      });
+  });
+  test("Should return a 404 and 'article_id does not exist' when given an invalid article_ID", () => {
+    return request(app)
+      .get("/api/articles/9001")
+      .expect(404)
+      .then(({ _body }) => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js:95]: why are you destructuring _body?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -46,15 +58,47 @@ describe("GET /api/topics", () => {
...
+      .then(({ body }) => {
+        let article = body.article;
+        expect(article.article_id).toBe(1);
+        expect(article.title).toEqual(expect.any(String));
+        expect(article.topic).toEqual(expect.any(String));
+        expect(article.author).toEqual(expect.any(String));
+        expect(article.body).toEqual(expect.any(String));
+        expect(article.created_at).toEqual(expect.any(String));
+        expect(article.votes).toEqual(expect.any(Number));
+      });
```
Comment (daviidbartlett) [inline __tests__/app.test.js:89]: You can afford to be more particular with your testing here. You know what article should come back

File: app.js [hunk 5]
```diff
@@ -1,15 +1,26 @@
...
+const { getTopics, getArticleByID } = require("./controllers/controllers.js");
 
 app.get("/api/topics", getTopics);
+app.get("/api/articles/:article_id", getArticleByID);
 
 app.all("/*", (req, res) => {
   res.status(404).send({ msg: "Invalid endpoint." });
 });
 
 app.use((err, req, res, next) => {
```
Comment (daviidbartlett) [inline app.js]: I'd consider extracting this into there own file

File: app.js [hunk 6]
```diff
@@ -1,15 +1,26 @@
...
+    res.status(err.status).send({ msg: err.msg });
+  }
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request" });
+  }
   console.log(err);
```
Comment (daviidbartlett) [inline app.js]: you're sending back to responses (Which will error) if this function invokes

**Pull request merged**